--- a/hblas.cabal	2016-09-08 10:35:42.405154257 -0400
+++ b/hblas.cabal	2016-09-08 13:07:26.029550693 -0400
@@ -118,29 +118,11 @@
   -- Other library packages from which modules are imported.
   build-depends:
         base             >= 4.6    &&  < 4.9,
-        storable-complex >= 0.2.0  &&  < 0.3.0,
-        primitive        >= 0.5    &&  < 0.6,
+        storable-complex >= 0.2.0,
+        primitive        >= 0.5,
         vector
 
-
-  if flag(OpenBLAS)
-    extra-libraries: openblas pthread
-
-  if flag(OpenBLAS)&& os(OSX)
-    extra-lib-dirs: /usr/local/lib
-
-  if os(OSX)&&!flag(OpenBLAS)
-      frameworks: Accelerate
-      -- extra-libraries: cblas lapack
-
-  if os(windows) && !flag(OpenBLAS)
-      extra-libraries: blas lapack
-
-  if !os(windows)&& !os(OSX) && !flag(OpenBLAS)
-    extra-libraries: blas lapack
-
-  if flag(CBLAS)
-    extra-libraries: cblas
+  pkgconfig-depends: blas, cblas, lapack
 
 
 
