--- a/src/afs/afs_fetchstore.c	2016-04-27 10:30:51.000000000 -0400
+++ b/src/afs/afs_fetchstore.c	2016-06-04 05:40:46.632365209 -0400
@@ -886,20 +886,20 @@
 
 static
 struct fetchOps rxfs_fetchUfsOps = {
-    rxfs_fetchMore,
-    rxfs_fetchUfsRead,
-    rxfs_fetchUfsWrite,
-    rxfs_fetchClose,
-    rxfs_fetchDestroy
+    .more = rxfs_fetchMore,
+    .read = rxfs_fetchUfsRead,
+    .write = rxfs_fetchUfsWrite,
+    .close = rxfs_fetchClose,
+    .destroy = rxfs_fetchDestroy
 };
 
 static
 struct fetchOps rxfs_fetchMemOps = {
-    rxfs_fetchMore,
-    rxfs_fetchMemRead,
-    rxfs_fetchMemWrite,
-    rxfs_fetchClose,
-    rxfs_fetchDestroy
+    .more = rxfs_fetchMore,
+    .read = rxfs_fetchMemRead,
+    .write = rxfs_fetchMemWrite,
+    .close = rxfs_fetchClose,
+    .destroy = rxfs_fetchDestroy
 };
 
 afs_int32
--- a/src/rxkad/rxkad_client.c	2016-04-27 10:30:51.000000000 -0400
+++ b/src/rxkad/rxkad_client.c	2016-06-05 01:36:06.533046590 -0400
@@ -68,21 +68,21 @@
 #endif /* max */
 
 static struct rx_securityOps rxkad_client_ops = {
-    rxkad_Close,
-    rxkad_NewConnection,	/* every new connection */
-    rxkad_PreparePacket,	/* once per packet creation */
-    0,				/* send packet (once per retrans.) */
-    0,
-    0,
-    0,
-    rxkad_GetResponse,		/* respond to challenge packet */
-    0,
-    rxkad_CheckPacket,		/* check data packet */
-    rxkad_DestroyConnection,
-    rxkad_GetStats,
-    0,
-    0,
-    0,
+    .op_Close = rxkad_Close,
+    .op_NewConnection = rxkad_NewConnection,	/* every new connection */
+    .op_PreparePacket = rxkad_PreparePacket,	/* once per packet creation */
+    .op_SendPacket = 0,				/* send packet (once per retrans.) */
+    .op_CheckAuthentication = 0,
+    .op_CreateChallenge = 0,
+    .op_GetChallenge = 0,
+    .op_GetResponse = rxkad_GetResponse,		/* respond to challenge packet */
+    .op_CheckResponse = 0,
+    .op_CheckPacket = rxkad_CheckPacket,		/* check data packet */
+    .op_DestroyConnection = rxkad_DestroyConnection,
+    .op_GetStats = rxkad_GetStats,
+    .op_SetConfiguration = 0,
+    .op_Spare2 = 0,
+    .op_Spare3 = 0,
 };
 
 /* To minimize changes to epoch, we set this Cuid once, and everyone (including
--- a/src/afs/LINUX/osi_misc.c	2016-06-05 01:59:02.658591593 -0400
+++ b/src/afs/LINUX/osi_misc.c	2016-06-05 01:59:34.199761210 -0400
@@ -143,7 +143,7 @@
     code = osi_lookupname_internal(name, followlink, &mnt, &dp);
     if (!code) {
 #if defined(D_PATH_TAKES_STRUCT_PATH)
-	afs_linux_path_t p = { mnt, dp };
+	afs_linux_path_t p = { .mnt = mnt, .dentry = dp };
 	path = d_path(&p, buf, buflen);
 #else
 	path = d_path(dp, mnt, buf, buflen);
